### PR TITLE
add slds log_prob test

### PR DESCRIPTION
**Pull request by martinjankowiak**

### CODE REVIEW

File: test/pyro/test_hmm.py [hunk 1]
```diff
@@ -303,3 +303,57 @@ def test_switching_linear_hmm_log_prob(exact, num_steps, hidden_dim, obs_dim, nu
...
+
+    assert actual_dist.batch_shape == expected_dist.batch_shape
+    assert actual_dist.event_shape == expected_dist.event_shape
+
+    data = obs_mvn.sample(expected_dist.batch_shape + (num_steps,))
+    assert data.shape == expected_dist.shape()
+    expected_log_prob = expected_dist.log_prob(data)
+    assert expected_log_prob.shape == expected_dist.batch_shape
+    actual_log_prob = actual_dist.log_prob(data)
+    assert_close(actual_log_prob, expected_log_prob, atol=0.3, rtol=None)
```
Comment (martinjankowiak) [inline test/pyro/test_hmm.py]: need to use a pretty big `atol`

**Comment by martinjankowiak** [comment]
this test now fails in a way that can't just be numerics. but is the test correct?

**Comment by fritzo** [comment]
Thanks for adding a failing test. I'm pretty busy preparing the time series tutorials this week, but we should schedule some time to pair debug.

**Comment by martinjankowiak** [comment]
great this now passes 💯

**Comment by fritzo** [comment]
Can you resolve merge conflicts?

**Comment by martinjankowiak** [comment]
done

**Pull request merged**